modules/@apostrophecms/piece-type/index.js

```diff
@@ -1060,6 +1060,84 @@ module.exports = {
+
+            await self.apos.migration.eachDoc({ type: self.name }, async doc => {
+              await self.apos.doc.db.removeOne({ _id: doc._id });
+
+              if (doc.aposDocId && !doc._id.endsWith('published') && !doc._id.endsWith('draft')) {
+                for (const locale of locales) {
+                  const lastPublishedAt = new Date();
+                  const newDraft = {
+                    ...doc,
+                    aposLocale: `${locale}:draft`,
+                    aposMode: 'draft',
+                    aposDocId: doc._id,
+                    _id: `${doc.aposDocId}:${locale}:draft`,
+                    lastPublishedAt
+                  };
+                  const newPublished = {
+                    ...doc,
+                    aposLocale: `${locale}:published`,
+                    aposMode: 'published',
+                    aposDocId: doc._id,
+                    _id: `${doc.aposDocId}:${locale}:published`,
+                    lastPublishedAt
+                  };
+                  await self.apos.doc.db.insertOne(newDraft);
+                  await self.apos.doc.db.insertOne(newPublished);
+                }
+              }
+            });
+
+            await self.apos.attachment.recomputeAllDocReferences();
+
+            console.log(`Done localizing module ${self.name}`);
+          } else {
+            throw new Error('Localized option not set to true, so the module cannot be localized.');
+          }
+        }
+      },
+
+      unlocalize: {
+        usage: 'Remove duplicate documents when a module has not "localized" and "autopublish" anymore.' +
+        '\nOptions are:' +
+        '\n- locale: if not set, it is the project\'s default locale' +
+        '\n- mode: by default, published' +
+        '\nExample: node app [moduleName]:unlocalize --mode=published --locale=en',
+        async task(argv) {
+          if (!self.options.autopublish && !self.options.localized) {
+            const locale = argv.locale || self.apos.i18n.defaultLocale;
+            const mode = argv.mode || 'published';
+            console.log(`Removing duplicated documents and updating ${mode} ones`);
+
+            await self.apos.migration.eachDoc({ type: self.name }, async doc => {
+              await self.apos.doc.db.removeOne({ _id: doc._id });
+              if (doc.aposDocId && doc.aposLocale === `${locale}:${mode}` && doc.aposMode === mode) {
+                const newDoc = {
+                  ...doc,
+                  aposLocale: undefined,
+                  aposMode: undefined,
+                  _id: doc.aposDocId
+                };
+                await self.apos.doc.db.insertOne(newDoc);
+              }
+            });
+            await self.apos.attachment.recomputeAllDocReferences();
+
+            console.log(`Done unlocalizing module ${self.name}`);
+          } else {
+            throw new Error('Autopublish or localized option are not false, so the module cannot be unlocalized.');
```

Comment:
Same suggestion to factor this check and throw to the top and not nest the whole thing. Otherwise though this task seems 💯